slkspec/xr_accessor.py

```diff
@@ -0,0 +1,134 @@
+from itertools import compress
+
+import dask
+import xarray as xr
+
+
+@xr.register_dataarray_accessor("stage")
+@xr.register_dataset_accessor("stage")
```

Comment:
I strongly recommend adding a `slk` namespace or something similar

```suggestion
@xr.register_dataarray_accessor("slk")
@xr.register_dataset_accessor("slk")
```

so the usage is `DataArray.slk.stage()`. Then there's no confusion about whether `stage` is a built-in function provided by Xarray, or an external one registered using an accessor.